### ISSUE TITLE
Add support for macos

### ISSUE THREAD

**Issue by Kariaro** [game-ci/unity-test-runner#211]
**Context**
I need to test that external libraries work in Windows, Linux and Macos.

Currently macos is not supported. That's why I'm making this issue

**Suggested solution**

Copy code from `game-ci/unity-builder` that handles darwin systems and use it in this repository

**Additional details**

I'm going to try make a PR for this issue myself and will clone this repository to start testing

**Comment by webbertakken** [comment]
Yup! We support this approach and accept contributions.

**Comment by Kariaro** [comment]
I'm currently looking at the generated docker images for macos but I don't see any here: https://github.com/game-ci/docker/tree/main/images/macos

Should I copy the `MacBuilder` concept from https://github.com/game-ci/unity-builder/blob/main/src/model/mac-builder.ts and apply it so that if `process.platform === 'darwin'` I jump to the `MacTestRunner` instead?

**Comment by davidmfinol** [comment]
That's right, there are no docker images for macos, so you would likely need to copy the relevant typescript code from unity-builder.

**Comment by MichaelGlaive** [comment]
Any news here?
Any help needed?

**Comment by izzhy** [comment]
Any news or workarounds to deal with that problem ?

**Comment by frostebite** [comment]
Heads up: v5.0.0-beta.1 (see #316) delegates test execution, licensing, and coverage handling entirely to game-ci/cli's extracted logic, replacing this repo's own `src/model/` implementation. That's a strong candidate to have changed the behavior reported here.

Leaving this open for now — once the thin wrapper merges, please retest against the new version and let us know if this is still reproducible. If so, we'll want to track it against game-ci/cli directly going forward.

**Comment by frostebite** [comment]
Closing for now since the underlying implementation this reported against is being replaced by the thin-wrapper migration. Please reopen (or file a fresh issue) if this still reproduces once you're on the new version — happy to pick it back up.
